velox/exec/tests/ArrowStreamTest.cpp

```diff
@@ -0,0 +1,188 @@
+      const auto& lastError = privateData()->lastError_;
+      return lastError.empty() ? nullptr : lastError.c_str();
+    }
+
+    void release() {
+      if (stream_->release == NULL) {
+        // C array stream is released.
+        return;
+      }
+      DCHECK_NE(privateData(), nullptr);
+      delete privateData();
+      // Mark stream as released.
+      stream_->release = NULL;
+    }
+
+    static int staticGetSchema(
+        struct ArrowArrayStream* stream,
+        struct ArrowSchema* outSchema) {
+      ExportedArrayStream self{stream};
+      return self.getSchema(outSchema);
+    }
+
+    static int staticGetNext(
+        struct ArrowArrayStream* stream,
+        struct ArrowArray* outArray) {
+      ExportedArrayStream self{stream};
+      return self.getNext(outArray);
+    }
+
+    static void staticRelease(struct ArrowArrayStream* stream) {
+      ExportedArrayStream{stream}.release();
+    }
+
+    static const char* staticGetLastError(struct ArrowArrayStream* stream) {
+      return ExportedArrayStream{stream}.getLastError();
+    }
+
+   private:
+    PrivateData* privateData() {
+      return reinterpret_cast<PrivateData*>(stream_->private_data);
+    }
+    const std::shared_ptr<ArrowReader>& reader() {
+      return privateData()->reader_;
+    }
+
+    std::shared_ptr<memory::MemoryPool> pool_;
+    struct ArrowArrayStream* stream_;
+  };
+
+  void ExportArrowStream(
+      std::shared_ptr<ArrowReader> reader,
+      struct ArrowArrayStream* out) {
+    out->get_schema = ExportedArrayStream::staticGetSchema;
+    out->get_next = ExportedArrayStream::staticGetNext;
+    out->get_last_error = ExportedArrayStream::staticGetLastError;
+    out->release = ExportedArrayStream::staticRelease;
+    out->private_data = new ExportedArrayStream::PrivateData{std::move(reader)};
+  }
+};
+
+TEST_F(ArrowStreamTest, basic) {
+  vector_size_t size = 1'000;
+  std::vector<RowVectorPtr> vectors;
+  for (int32_t i = 0; i < 3; ++i) {
+    vectors.push_back(makeRowVector(
+        {makeFlatVector<bool>(
+             size, [](auto row) { return (row / 10) % 2 == 0; }, nullEvery(3)),
+         makeFlatVector<int8_t>(
+             size, [](auto row) { return row % 100; }, nullEvery(5)),
+         makeFlatVector<int16_t>(
+             size, [&](auto row) { return row * i; }, nullEvery(2)),
+         makeFlatVector<int32_t>(
+             size, [](auto row) { return row; }, nullEvery(5)),
+         makeFlatVector<int64_t>(
+             size,
+             [&](vector_size_t row) { return size * i + row; },
+             nullEvery(5)),
+         makeFlatVector<float>(
+             size, [](vector_size_t row) { return row * 1.1; }, nullEvery(5)),
+         makeFlatVector<double>(
+             size, [](vector_size_t row) { return row * 1.3; }, nullEvery(11)),
+         makeFlatVector<StringView>(
+             size,
+             [](vector_size_t row) {
+               return StringView(std::to_string(row % 100));
```

Comment:
Can we also test with strings longer that 12 characters?

Reply:
Fixed by using 12 characters as prefix, is it OK?